repos/system_upgrade/el7toel8/actors/tcpwrappersconfigread/libraries/library.py

```diff
@@ -0,0 +1,57 @@
+import errno
+import re
+
+from leapp.libraries.stdlib import api
+from leapp.models import DaemonList, TcpWrappersFacts
+
+CONFIG_FILES = ['/etc/hosts.allow', '/etc/hosts.deny']
+
+
+def read_file(path):
+    with open(path, 'r') as f:
+        return f.read()
+
+
+def get_lines(content):
+    content = re.sub(r'\\\n', '', content)
+    return content.split('\n')
+
+
+def is_comment(line):
+    return len(line) == 0 or line.isspace() or line.startswith('#')
+
+
+def get_daemon_list_in_line(line):
+    daemon_list = line.split(':', 1)[0]
+    daemon_list = daemon_list.replace(',', ' ')
+    daemon_list = daemon_list.replace('\t', ' ')
+    daemon_list = daemon_list.split(' ')
+    daemon_list = [word for word in daemon_list if len(word) > 0]
+    return DaemonList(value=daemon_list)
+
+
+def get_daemon_lists_in_file(path, read_func=read_file):
+    ret = []
+    try:
+        content = read_func(path)
+    except IOError as e:
+        if e.errno != errno.ENOENT:
+            api.current_logger().warning('Failed to read %s: %s' % (path, e))
+        return ret
+    lines = get_lines(content)
+    lines = [line for line in lines if not is_comment(line)]
```

Comment:
nit: Squash L41-L45?
```
return [get_daemon_list_in_line(line) for line in get_lines(content) if not is_comment(line)]
```

Reply:
OK. I've learned to instinctively avoid complex list comprehensions, that's why you're seeing this :). But this is just fine. Done.